lib/mix/test/mix/tasks/test_test.exs

```diff
@@ -408,6 +408,26 @@ defmodule Mix.Tasks.TestTest do
     end
   end
 
+  describe "--warnings-as-errors" do
+    test "fail on warning in tests" do
+      in_fixture("test_stale", fn ->
+        File.write!("test/warning_test.exs", """
```

Comment:
Let's also write a file in lib with:

    unused_compile = 13

And we will check it warns but still run tests. :)